homeassistant/components/meteoalarm/config_flow.py

```diff
@@ -0,0 +1,114 @@
+"""MeteoAlarm config flow."""
+
+from typing import Any
+
+from meteoalertapi import Meteoalert
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_COUNTRY,
+    CONF_LANGUAGE,
+    CONF_PROVINCE,
+    DEFAULT_COUNTRY,
+    DOMAIN,
+    LOGGER,
+    SUPPORTED_COUNTRIES,
+)
+
+SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_COUNTRY, default=DEFAULT_COUNTRY): selector.CountrySelector(
+            selector.CountrySelectorConfig(countries=list(SUPPORTED_COUNTRIES.keys()))
+        ),
+        vol.Required(CONF_PROVINCE): cv.string,
+        vol.Optional(CONF_LANGUAGE, default="en"): cv.string,
+    }
+)
+
+
+class MeteoAlarmConfigFlow(ConfigFlow, domain=DOMAIN):
+    """MeteoAlarm config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle config flow initiated by the user."""
+        errors = {}
+        if user_input:
+            self._async_abort_entries_match(
+                {
+                    CONF_COUNTRY: SUPPORTED_COUNTRIES.get(
+                        user_input[CONF_COUNTRY], DEFAULT_COUNTRY
+                    ),
+                    CONF_PROVINCE: user_input[CONF_PROVINCE],
+                    CONF_LANGUAGE: user_input[CONF_LANGUAGE],
+                }
+            )
+
+            try:
+                Meteoalert(CONF_COUNTRY, CONF_PROVINCE, CONF_LANGUAGE)
+
+            except KeyError:
+                LOGGER.error("Wrong country digits or province name")
+                errors["base"] = "wrong_country_or_province"
```

Comment:
Country can't be wrong now right?

Is there a predefined list of provinces per country? Maybe it's worth introducing an extra step to select the province so people can't mess it up (as in, if we can avoid people looking up what to fill in and make mistakes, that's always nice)

Reply:
I had a look some time ago into this integration and it simply isn't possible to fetch the list dynamically at runtime. 

As far as I'm aware, there's also no KeyError being throw in the library to know that the province provided is incorrect. When the wrong province is given, it will simply return no alerts (since it will never match your wrong province.) 

The list of countries is provided on their https://feeds.meteoalarm.org/ page only, so hardcoding is the only solution here (no html parsing allowed). The list of provinces is also available in a Google docs json file available through their redistribution hub. That's a 30MB file with shapes included. I think hardcoding is also the only solution here, but note that some countries do have a lot of provinces.

Reply:
This is a good one to check with others. I think it's nice to make it foolproof, because you also explain it won't work otherwise.

But that dict should ideally live in the library